api/http/security/bouncer.go

```diff
@@ -528,7 +529,12 @@ func (bouncer *RequestBouncer) EdgeComputeOperation(next http.Handler) http.Hand
 // - public routes
 // - kubectl - a bearer token is needed, and no csrf token can be sent
 // - api token
+// - docker desktop extension
 func ShouldSkipCSRFCheck(r *http.Request) (bool, error) {
+	if _, ok := os.LookupEnv("DOCKER_EXTENSION"); ok {
```

Comment:
Do we really need to disable CSRF checks for the Docker extension?

Reply:
It seems so, I wanted to remove it, but from testing, when I restarted docker desktop and started it again with the portainer extension, I got an auth error when when portainer tried to auto login.

Reply:
Thinking about that if CSRF only impacts auto login, it could indicate some weaknesses in the business logic. Instead of bypassing the CSRF check using the DOCKER_EXTENSION environment variable, we could address the CSRF issue and guide the user to the login page based on the environment variable. 
For instance, with the docker extension, we could implement a separate auto login/login process to maintain the CSRF feature.
But when considering the time needed to implement this for version 2.21, I think the current solution may be a compromise.

Reply:
Good summary. I would not want to create too many logical paths, especially one for the DD extension if we can avoid it.

I think that LP's suggestion in Linear is actually the best way to approach this. We should lookup for the DOCKER_EXTENSION environment variable at startup only, store it in memory and pass a flag down to the bouncer.

I would not want a user being able to disable CSRF checks at runtime by injecting an environment variable in the container.

Reply:
Good call, updated